community/cypher/cypher/src/main/java/org/neo4j/cypher/internal/javacompat/QueryEngineProcedures.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.javacompat;
+
+import java.util.stream.Stream;
+
+import org.neo4j.graphdb.security.AuthorizationViolationException;
+import org.neo4j.kernel.api.security.SecurityContext;
+import org.neo4j.kernel.impl.query.QueryExecutionEngine;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
+import org.neo4j.procedure.Context;
+import org.neo4j.procedure.Description;
+import org.neo4j.procedure.Procedure;
+
+import static org.neo4j.graphdb.security.AuthorizationViolationException.PERMISSION_DENIED;
+import static org.neo4j.procedure.Mode.DBMS;
+
+public class QueryEngineProcedures
+{
+    @Context
+    public GraphDatabaseAPI graph;
+
+    @Context
+    public SecurityContext context;
+
+    @Description( "Clears all query caches." )
+    @Procedure( name = "dbms.clearQueryCaches", mode = DBMS )
+    public Stream<StringResult> clearAllQueryCaches()
+    {
```

Comment:
Why not place this procedure in the `BuiltInDbmsProcedures`?

Reply:
Same reason, I cannot move the procedure registration - I would create a dependency from Kernel (´BuiltInDbmsProcedures´) to Cypher (´QueryExecutionEngine´)